src/core/mac/mac.hpp

```diff
@@ -647,6 +647,22 @@ class Mac: public ThreadNetifLocator
      */
     bool RadioSupportsRetries(void);
 
+    /**
+     * This method returns the amount of time the radio has spent in Tx mode.
+     *
+     * @returns  A pointer to the total Tx time.
+     *
+     */
+    uint32_t GetRadioTxTotalTime(void) { return mTxTotal; }
+
+    /**
+     * This method returns the amount of time the radio has spent in Rx mode.
+     *
+     * @returns  A pointer to the total Rx time.
+     *
+     */
+    uint32_t GetRadioRxTotalTime(void) { return mRxTotal; }
```

Comment:
If the mRadioState == kRadioStateRx at the time this call is made then there exists some amount of unaccounted rx time by this call.  Conceivably the radio could be in Rx for a quite some time and this discrepancy could then become a significant percentage of the total.  What if this function were written as;

return (mRadioState == kRadioStateRx) ? mRxTotal + TimerMilli::GetNow() - mLastChange : mRxTotal;

Similar logic could exist for GetRadioTxTotalTime().